app/screens/home/channel_list/categories_list/categories_list.tsx

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useEffect, useMemo} from 'react';
+import {useWindowDimensions} from 'react-native';
+import Animated, {useAnimatedStyle, useSharedValue, withTiming} from 'react-native-reanimated';
+
+import DraftsButton from '@components/drafts_buttton';
+import ThreadsButton from '@components/threads_button';
+import {TABLET_SIDEBAR_WIDTH, TEAM_SIDEBAR_WIDTH} from '@constants/view';
+import {useTheme} from '@context/theme';
+import {useIsTablet} from '@hooks/device';
+import {makeStyleSheetFromTheme} from '@utils/theme';
+
+import Categories from './categories';
+import ChannelListHeader from './header';
+import LoadChannelsError from './load_channels_error';
+import SubHeader from './subheader';
+
+const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => ({
+    container: {
+        flex: 1,
+        backgroundColor: theme.sidebarBg,
+        paddingTop: 10,
+    },
+}));
+
+type ChannelListProps = {
+    hasChannels: boolean;
+    iconPad?: boolean;
+    isCRTEnabled?: boolean;
+    moreThanOneTeam: boolean;
+    currentChannelId: string;
+    draftsCount: number;
+};
+
+const getTabletWidth = (moreThanOneTeam: boolean) => {
+    return TABLET_SIDEBAR_WIDTH - (moreThanOneTeam ? TEAM_SIDEBAR_WIDTH : 0);
+};
+
+const CategoriesList = ({
+    hasChannels,
+    iconPad,
+    isCRTEnabled,
+    moreThanOneTeam,
+    currentChannelId,
+    draftsCount,
+}: ChannelListProps) => {
+    const theme = useTheme();
+    const styles = getStyleSheet(theme);
+    const {width} = useWindowDimensions();
+    const isTablet = useIsTablet();
+    const tabletWidth = useSharedValue(isTablet ? getTabletWidth(moreThanOneTeam) : 0);
+
+    useEffect(() => {
+        if (isTablet) {
+            tabletWidth.value = getTabletWidth(moreThanOneTeam);
+        }
+    }, [isTablet, moreThanOneTeam]);
+
+    const tabletStyle = useAnimatedStyle(() => {
+        if (!isTablet) {
+            return {
+                maxWidth: width,
+            };
+        }
+
+        return {maxWidth: withTiming(tabletWidth.value, {duration: 350})};
+    }, [isTablet, width]);
+
+    const content = useMemo(() => {
+        if (!hasChannels) {
+            return (<LoadChannelsError/>);
+        }
+
+        return (
+            <>
+                <SubHeader/>
+                {isCRTEnabled &&
+                    <ThreadsButton
+                        isOnHome={true}
+                        shouldHighlighActive={true}
+                    />
+                }
+                {draftsCount > 0 && (
+                    <DraftsButton
+                        draftsCount={draftsCount}
+                        currentChannelId={currentChannelId}
+                    />
+                )}
+                <Categories/>
+            </>
+        );
+    }, [currentChannelId, draftsCount, hasChannels, isCRTEnabled]);
```

Comment:
Is there a huge benefit of memoizing this?  Wouldn't even draftsCount change would cause the entire content to be recreated? Would React default diff algorithm be efficient enough to handle the changes.  If draftsCount change, would it create a new memoized output or use the standard diffing on render?

Reply:
React diffing algorithm knows what to change.  `content` gets re-assigned with JSX.element on any of the dependency changes.  So as an example, if draftsCount change, would ThreadsButton gets re-rendered as well?  Whereas React diffing would know not to do that.

Thus the reason of my query.  If React diffing works in useMemo as well, then I think we're all good here.  I would think so, but I'm not sure myself.

Reply:
I was curious so I created a test myself:

https://codesandbox.io/p/sandbox/relaxed-shadow-5jskx4

The diffing works within useMemo.  As suspected.  But again, the React diffing algorithm should be able to handle this on it's own.  As changes to draftsCount would not cause ThreadsButton to re-render, similar to useMemo.  You can modify the same demo to mimic what you're doing.

Reply:
So I have created a test demo by modifying your changes.

https://codesandbox.io/p/sandbox/funny-diffie-hc88fj 

In the demo, I added a `console.log("Re-rendering")` inside `SomeOtherComponent` to verify if it re-renders. Upon clicking the "Press Me" button, the console logs `Re-rendering`, indicating that `SomeOtherComponent` is indeed re-rendering.

I also tested by removing the useMemo to see if React's diffing algorithm handles the re-rendering on its own. However, it doesn’t seem to prevent `SomeOtherComponent` from re-rendering in this case.

Let me know if you see anything I might be missing!